src/www/app/outline_server.ts

```diff
@@ -223,45 +219,88 @@ export class OutlineServerRepository implements ServerRepository {
   }
 
   private storeServers() {
-    const configByAccessKey: AccessKeyById = {};
-    for (const [serverId, server] of this.serverById) {
-      configByAccessKey[serverId] = server.accessKey;
+    const servers: ServersStorageV1 = [];
+    for (const server of this.serverById.values()) {
+      servers.push({
+        id: server.id,
+        accessKey: server.accessKey,
+        name: server.name,
+      });
     }
-    const json = JSON.stringify(configByAccessKey);
+    const json = JSON.stringify(servers);
     this.storage.setItem(OutlineServerRepository.SERVERS_STORAGE_KEY, json);
   }
 
   // Loads servers from storage, raising an error if there is any problem loading.
   private loadServers() {
+    if (this.storage.getItem(OutlineServerRepository.SERVERS_STORAGE_KEY)) {
+      console.debug('server storage migrated to V1');
+      this.loadServersV1();
+      return;
+    }
+    this.loadServersV0();
+  }
+
+  private loadServersV0() {
     this.serverById = new Map<string, OutlineServer>();
-    const serversJson = this.storage.getItem(OutlineServerRepository.SERVERS_STORAGE_KEY);
+    const serversJson = this.storage.getItem(OutlineServerRepository.SERVERS_STORAGE_KEY_V0);
     if (!serversJson) {
-      console.debug(`no servers found in storage`);
+      console.debug(`no V0 servers found in storage`);
       return;
     }
-    let configByAccessKey: AccessKeyById = {};
+    let configById: ServersStorageV0 = {};
     try {
-      configByAccessKey = JSON.parse(serversJson);
+      configById = JSON.parse(serversJson);
     } catch (e) {
-      throw new Error(`could not parse saved servers: ${e.message}`);
+      throw new Error(`could not parse saved V0 servers: ${e.message}`);
     }
-    for (const serverId in configByAccessKey) {
-      if (configByAccessKey.hasOwnProperty(serverId)) {
-        const accessKey = configByAccessKey[serverId];
+    for (const serverId in configById) {
```

Comment:
No need to change, but I think `for (const serverId of Object.keys(configById))` is the right way to do this. It removes the need for the hasOwnProperty below, I believe.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys

Reply:
Done in 3e18673.